drivers/vultr/vultr.go

```diff
@@ -0,0 +1,362 @@
+	d := &Driver{MachineName: machineName, storePath: storePath, CaCertPath: caCert, PrivateKeyPath: privateKey, bucket: nil}
+	// throttle API queries according to 1/s rate-limit
+	d.bucket = tokenbucket.NewBucket(1*time.Second, 1)
+	return d, nil
+}
+
+func (d *Driver) AuthorizePort(ports []*drivers.Port) error {
+	return nil
+}
+
+func (d *Driver) DeauthorizePort(ports []*drivers.Port) error {
+	return nil
+}
+
+func (d *Driver) GetMachineName() string {
+	return d.MachineName
+}
+
+func (d *Driver) GetSSHHostname() (string, error) {
+	return d.GetIP()
+}
+
+func (d *Driver) GetSSHKeyPath() string {
+	return filepath.Join(d.storePath, "id_rsa")
+}
+
+func (d *Driver) GetSSHPort() (int, error) {
+	if d.SSHPort == 0 {
+		d.SSHPort = 22
+	}
+
+	return d.SSHPort, nil
+}
+
+func (d *Driver) GetSSHUsername() string {
+	if d.SSHUser == "" {
+		d.SSHUser = "root"
+	}
+
+	return d.SSHUser
+}
+
+func (d *Driver) GetProviderType() provider.ProviderType {
+	return provider.Remote
+}
+
+func (d *Driver) DriverName() string {
+	return "vultr"
+}
+
+func (d *Driver) SetConfigFromFlags(flags drivers.DriverOptions) error {
+	d.APIKey = flags.String("vultr-api-key")
+	d.OSID = flags.Int("vultr-os-id")
+	d.RegionID = flags.Int("vultr-region-id")
+	d.PlanID = flags.Int("vultr-plan-id")
+	d.IPv6 = flags.Bool("vultr-ipv6")
+	d.PrivateNetworking = flags.Bool("vultr-private-networking")
+	d.Backups = flags.Bool("vultr-backups")
+	d.SwarmMaster = flags.Bool("swarm-master")
+	d.SwarmHost = flags.String("swarm-host")
+	d.SwarmDiscovery = flags.String("swarm-discovery")
+	d.SSHUser = "root"
+	d.SSHPort = 22
+
+	if d.APIKey == "" {
+		return fmt.Errorf("vultr driver requires the --vultr-api-key option")
+	}
+	return nil
+}
+
+func (d *Driver) PreCreateCheck() error {
+
+	if err := d.validateRegion(); err != nil {
+		return err
+	}
+
+	if err := d.validatePlan(); err != nil {
+		return err
+	}
+
+	if err := d.validateApiCredentials(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *Driver) Create() error {
+	log.Infof("Creating SSH key...")
+	key, err := d.createSSHKey()
+	if err != nil {
+		return err
+	}
+	d.SSHKeyID = key.ID
+
+	log.Infof("Creating Vultr virtual machine...")
+
+	client := d.getClient()
+	machine, err := client.CreateServer(
+		d.MachineName,
+		d.RegionID,
+		d.PlanID,
+		d.OSID,
+		&vultr.ServerOptions{
+			SSHKey:            d.SSHKeyID,
+			IPV6:              d.IPv6,
+			PrivateNetworking: d.PrivateNetworking,
+			AutoBackups:       d.Backups,
+		})
+	if err != nil {
+		return err
+	}
+	d.MachineID = machine.ID
+
+	for {
+		log.Debugf("Getting server details...")
+		<-d.bucket.SpendToken(1)
+		machine, err = client.GetServer(d.MachineID)
+		if err != nil {
+			return err
+		}
+		d.PublicIP = machine.MainIP
+		d.PrivateIP = machine.InternalIP
+
+		if d.PublicIP != "" && d.PublicIP != "0" {
+			break
+		}
+	}
+
+	if d.PrivateIP == "" || d.PrivateIP == "0" {
+		d.PrivateIP = "N/A"
```

Comment:
Any reason this is `N/A` instead of empty string? The convention in other drivers is that it gets set to `""`